viewing_party/party.py

```diff
@@ -1,23 +1,248 @@
+            watched_genre_list.append(user_data["watched"][i]["genre"])
+        
+        most_watched_count = 0
+        most_watched_genre = ''
+
+        # iterate through genre list, counting how many times each genre appears
+        for genre in watched_genre_list:
+            genre_count = watched_genre_list.count(genre)
+            # update most_watched_genre and genre count with whatever comes back highest
+            if genre_count > most_watched_count:
+                most_watched_count = genre_count
+                most_watched_genre = genre
+
+        return most_watched_genre
+    else:
+        return None
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    """
+    Creates list of movies that user has watched but friends haven't.
+
+    Parameters: user data, including user's watched list, and friends watched lists
+
+    Returns: new list of movie information dictionaries
+    """
+    # create empty list to store unique movies
+    user_unique_watched = []
+
+    # iterate over user's watched list
+    for i in range(len(user_data["watched"])):
+        # set flag to signal if we're still looking
+        searching_for_movie = True
+
+        # set current title we're looking for
+        looking_for_title = user_data["watched"][i]["title"]
+
+        while searching_for_movie:
+            # iterate through friends list
+            for friend in user_data["friends"]:
+                friend_index = user_data["friends"].index(friend)
+                # iterate through every movie on friend's watch list
+                if searching_for_movie:
+                    for num in range(len(user_data["friends"][friend_index]["watched"])):
+                        # if title matches title we're looking for, break out of loop and start looking for next title
+                        # once a title has been found once, we don't need to see if it's been watched by multiple friends
+                        if user_data["friends"][friend_index]["watched"][num]["title"] == looking_for_title:
+                            searching_for_movie = False
+                            break
+                        else:
+                            continue
+                # if no longer searching for movie, break out of loops until searching movie is changed
+                elif not searching_for_movie:
+                    break
+            # if after iterating through all friends watched lists we are still looking, add movie to the unique list
+            if searching_for_movie:
+                user_unique_watched.append(user_data["watched"][i])
+                # then change flag to switch to next movie
+                searching_for_movie = False
+        # continue until all movies on user's watch list have been checked. 
+        continue
+
+    return user_unique_watched       

```

Comment:
I'd like to chat about this function in our next 1-1 to hear about your approach. We're looping over a fixed number of items, so we could solve this without a `while` loop. One way would be to create a list that contains all the movies in all our friends list, then loop over `user_data["watched"]` and if a movie is not in our list of friend movies, add it to our unique list.  
```
    friend_movies = []
    #get friends movies
    for watched_dict in user_data['friends']:
        for movie in watched_dict['watched']:
            if movie not in friend_movies:
                friend_movies.append(movie)

    user_unique_watched = []
    #check for unique titles
    for movie in user_data['watched']:
        if movie not in friend_movies:
            user_unique_watched.append(movie)
    
    return user_unique_watched
```

Reply:
This was definitely the function I struggled with the most!  I found it way harder than the other waves. I think I was over-complicating it, in retrospect. But yes, lets talk it over at our next meeting (assuming nothing else urgent has come up)!